### ISSUE TITLE
Initial focus is not set/visible when the Modal is opened and focus order is not same as the visual order

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4600]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 2.4.3 Focus Order

**Test Environment:**
- Test URL: https://clarity.design/documentation/modals
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 19.3.1

**Steps to reproduce:**
1. Open the test URL.
2. Move focus onto to the first Show Modal button present under Angular modals.
3. Activate the show modal button using enter or space key and observe the focus by navigating using tab key.

**Expected result:**
Focus should be placed on the first interactive element and tab order should follow visual or logical order as applicable.

**Reference:**
https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html
Content from the reference:  A Web page implements modal dialogs via scripting. When the trigger button is activated, a dialog opens and focus is set to the first interactive element in the dialog. As long as the dialog is open, focus is limited to the elements of the dialog. When the dialog is dismissed, focus returns to the button or the element following the button.

**Actual result:**
Focus is not visible or missing. If the user uses the tab key navigation, focus order is:
1. Body content.
2. Cancel.
3. Ok.
4. Overall dialog.
5. Close button.
The navigation order is not logical. For sighted users, the Ok button is highlighted, so the intent is to attract the users to press the ok button, same importance is not given to Ok button when using keyboard navigation.

**Proposed solution:**
Place initial focus on the Ok button, then tab order should be:
1. Cancel button.
2. Close button.

**Note:** Overall dialog doesn&#039;t need to have a separate focus.

**Reporter Email:** gkhadke@vmware.com

**Comment by gnomeontherun** [comment]
This is against what was decided with @chlane where the focus is put onto the title. It is intentional then so that the title is read immediately on open.

Closing as intended.

**Comment by chlane** [comment]
Yes. This is an intentional focus management decision which maintains a logical reading order. Regarding the lack of the visible focus outline on the heading, this is design decision that I recommended against but was overruled on. But it has low user impact. Sighted keyboard-only users may lose track of the focused element momentarily but will see the focus outline again the next time they press the Tab key.

**Comment by gkhadK** [comment]
I think logical order as per WCAG is to place the focus on the first interactive element. Screen reader will should still read the dialog title and body content and at the end it should tell the user about the currently focused item.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by chlane** [comment]
Focus management should be considered case by case. Often the close button is the first interactive element in the DOM order and if it receives focus first the user is taken to the exit before they start reading the content. I feel focus should go where the designer intends the user's eyes should go. 

I realize this is different than what is recommended at https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html but it is accessible. Screen reader users will not miss any content with this approach.

**Comment by gkhadK** [comment]
I agree there is no information loss, but it adds an extra navigation step for keyboard as well as screen reader user.

Thanks,
Guru.
________________________________

**Comment by chlane** [comment]
@gkhadK and @gnomeontherun I think we are missing the spirit of the bug here. While I did recommend we move focus to the static heading I did not recommend suppressing the visual focus indication. So my recommendation is to provide visual focus indication for all elements that gain focus, even headings.  If this is not possible, the solution @gkhadK provides is better.

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
